llvm/test/Transforms/InstCombine/shl-factor.ll

```diff
@@ -265,3 +265,27 @@ define i6 @sub_shl_same_amount_partial_nuw2(i6 %x, i6 %y, i6 %z) {
   ret i6 %diff
 }
 
+define i8 @add_shl_same_amount_constants(i8 %z) {
```

Comment:
Can you pre-commit these tests? See also https://llvm.org/docs/InstCombineContributorGuide.html#precommit-tests.

Reply:
These new tests aren't actually affected by this patch so there will be no diff, I added them since they show a pattern that wasn't tested before that I want to show as being preserved despite the restrictions added by the change. Should I still pre-commit? Or maybe these should be added in a separate NFC PR?

Reply:
Yeah, please file a separate PR for these additional tests.

Since this patch is not NFC, you should pre-commit some tests to demonstrate the issue.

Reply:
Removed the unaffected new tests and added a pre-commit to demonstrate the issue with `tryFactorization` transform. The issue with `factorizeMathWithShlOps` is already shown by the changes to shl-factor.ll.